SensusService/Protocol.cs

```diff
@@ -300,14 +301,14 @@ public static void DisplayAndStartAsync(Protocol protocol)
                         else
                         {
                             protocolsPage = new ProtocolsPage();
-                            await Application.Current.MainPage.Navigation.PushAsync(protocolsPage);
+                            SensusServiceHelper.Get().MainThreadSynchronizer.ExecuteThreadSafe(protocolsPage.Bind);
```

Comment:
This looks wrong. The page was just created, and it needs to be pushed on to the navigation stack. Rebinding won't do anything.